Environment/Puzzles/SwitchDoor.cs

```diff
@@ -0,0 +1,62 @@
+using Godot;
+namespace CrossDimensions.Environment.Puzzles;
+public partial class SwitchDoor : Node
+{
+    [Export]
+    public Godot.Collections.Array<Node2D> Switches { get; set; } = [];
+    private Godot.Collections.Array<SwitchButton> Buttons { get; set; } = [];
```

Comment:
```suggestion
    private Godot.Collections.Array<SwitchButton> _buttons = [];
```

getters and setters not needed for private members, so direct access would be fine